### ISSUE TITLE
Graphs not rendering in PDF when using dev.new() with knitr

### ISSUE THREAD

**Issue by Higgs32584** [yihui/knitr#2355]
![Screenshot 2024-07-09 120145](https://github.com/yihui/knitr/assets/55243596/81a0b1f7-2179-44e3-a2e1-876fc2706ca0)
![Screenshot 2024-07-09 120128](https://github.com/yihui/knitr/assets/55243596/4948da36-0617-4870-96fb-4600a89b68ba)
![Screenshot 2024-07-09 120101](https://github.com/yihui/knitr/assets/55243596/91a7826d-de4d-49f5-aa68-45865d6fcc67)

Hi, so recently, I was using knitr to compile graphs for my homework and, I was adding dev.new() before all of my graphs, as I thought this was the normal format to create graphs in.
What did end up happening though, is that graphs that followed dev.new() did not render at all in the pdf. while I understand dev.new is a function in the graphics package, I ran into no issues when compiling the graphs in markdown, as they all displayed perfectly fine, the code interaction seems to be related specifically towards how knitr compiles  .rmd into a .pdf.

A warning label related to instances of dev.new() in the compilation would be ideal, as I imagine taking into account new graph windows might be too complex. 

```{r setup, include=FALSE}
library("robustbase")
data("coleman",package="robustbase")
```
```{r}
dev.new()
barplot(salaryP ~ fatherWc,data=coleman)
```

```{r}
dev.new()
plot(salaryP ~ fatherWc,data=coleman)
```

**Comment by cderv** [comment]
Hi. 

Can you share you session information please ? 

I wonder if this is linked to an update in the **evaluate** package for 0.24

**Comment by cderv** [comment]
@yihui I am not sure if we made changes recently that would have caused this... I tried different version with no success. 

You know better than me how `dev.new()` works in chunks. My understanding is that it is not needed, but shouldn't behave like it is here.

**Comment by Higgs32584** [comment]
![Screenshot 2024-07-11 103307](https://github.com/yihui/knitr/assets/55243596/16983ec5-44f6-45a7-b93b-49d346d3ed03)

**Comment by yihui** [comment]
Thanks for the report. The root cause is that `dev.new()` opens a *new* graphics device inside the chunk, which becomes the current device. knitr draws and records plots on its own device (created for the chunk with plot recording enabled via `dev.control(displaylist = 'enable')`). The device from `dev.new()`, however, has display list recording turned off by default, so `recordPlot()` returns an empty plot and knitr captures nothing — hence the missing figures. This is independent of the output device (`png`, `pdf`, etc.) and not related to any change in the **evaluate** package.

In short, `dev.new()` is not needed in a knitr chunk (knitr manages the device for you), and using it will prevent your plots from being captured. As requested, knitr now emits a warning when a chunk opens a new graphics device, e.g.:

> Warning message:
The chunk 'unnamed-chunk-1' opened new graphics device(s) (e.g., via dev.new()), from which knitr cannot capture plots. Please remove the code that opens new devices such as dev.new().

Just remove the `dev.new()` calls and the plots will render as expected. Fixed in the current development version.
